src/main/java/com/seatapp/controllers/SeatsController.java

```diff
@@ -102,14 +112,40 @@ public ResponseEntity<List<Seat>> getSeatsWithReservationsByDate(
      *
      * @param seatId         the Id of the to be reserved seat.
      * @param reservationDto the reservation details.
+     * @param token          the authentication information.
      * @return Returns a response with the HttpStatus and a message.
      */
     @PatchMapping("{seatId}/reserve")
-    public ResponseEntity<String> reserveSeat(@PathVariable final Long seatId,
-                                              @RequestBody final
-                                              ReservationDto reservationDto) {
-        Seat reservedSeat = seatService.reserve(seatId, reservationDto);
-        return ResponseEntity.status(HttpStatus.OK)
-                .body("You reserved " + reservedSeat.getName() + ".");
+    public ResponseEntity<String>
+    reserveSeat(@PathVariable final Long seatId,
+                @RequestBody final
+                ReservationDto reservationDto,
+                final
+                UsernamePasswordAuthenticationToken
+                        token) {
+        User user = userService.getByUsername(token.getName());
+        seatService.reserve(seatId, reservationDto, user);
+        return ResponseEntity.ok().build();
+    }
+
+    /**
+     * Checks in on an existing seat.
+     *
+     * @param seatId the Id of the to be checked in seat.
+     * @param token  the authentication information.
+     * @return Returns a response with the HttpStatus and a message.
+     */
+    @PatchMapping("{seatId}/checkIn")
+    public ResponseEntity<String> checkIn(@PathVariable final Long seatId,
+                                          final
+                                          UsernamePasswordAuthenticationToken
+                                                  token) {
+        if (userService.existsByUsername(token.getName())) {
+            seatService
+                    .checkInOnSeat(seatId, token.getName());
+            return ResponseEntity.ok().build();
+        } else {
+            throw new EntityNotFoundException("This user doesn't exist");
```

Comment:
Don't throw exception in the controller class. You can better return a ResponseEntity.